pkg/broker/broker_test.go

```diff
@@ -64,3 +65,176 @@ func ExampleBrokerService_IsEnabled() {
 	// Output: true
 	// false
 }
+
+func ExampleBrokerService_ServiceDefinition() {
+	service := BrokerService{
+		Name: "left-handed-smoke-sifter",
+		DefaultServiceDefinition: `{"id":"abcd-efgh-ijkl"}`,
+	}
+
+	// Default definition
+	defn, err := service.ServiceDefinition()
+	fmt.Printf("%q %v\n", defn.ID, err)
+
+	// Override
+	viper.Set(service.DefinitionProperty(), `{"id":"override-id"}`)
+	defn, err = service.ServiceDefinition()
+	fmt.Printf("%q %v\n", defn.ID, err)
+
+	// Bad Value
+	viper.Set(service.DefinitionProperty(), "nil")
+	_, err = service.ServiceDefinition()
+	fmt.Printf("%v\n", err == nil)
+
+	// Cleanup
+	viper.Set(service.DefinitionProperty(), nil)
+
+	// Output: "abcd-efgh-ijkl" <nil>
+	// "override-id" <nil>
+	// false
+}
+
+func ExampleBrokerService_GetPlanById() {
+	service := BrokerService{
+		Name: "left-handed-smoke-sifter",
+		DefaultServiceDefinition: `{"id":"abcd-efgh-ijkl", "plans": [{"id": "builtin-plan", "name": "Builtin!"}]}`,
+	}
+
+	viper.Set(service.UserDefinedPlansProperty(), `[{"id":"custom-plan", "name": "Custom!"}]`)
+	defer viper.Set(service.UserDefinedPlansProperty(), nil)
+
+	plan, err := service.GetPlanById("builtin-plan")
+	fmt.Printf("%q %v\n", plan.Name, err)
```

Comment:
It could be helpful for reading the output to prefix each Printf with the use-case being demonstrated, eg: fmt.Printf("builtin-plan: %q %v\n")